packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -50,6 +54,31 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
             additionalProps[header.logName] = headerValue;
         }
     });
+
+    // x-fluid-telemetry contains a key value pair in the following format:
+    // x-fluid-telemetry:key1=value1,key2,key3=value3,
+    // Ex. x-fluid-telemetry:Origin=c,isSomeDataPoint (there will be no isSomeDataPoint=false or isSomeDataPoint=True)
+    const fluidTelemetry = headers.get("x-fluid-telemetry");
+    if (fluidTelemetry !== undefined && fluidTelemetry !== null) {
+        const mapResponseOrigin: Map<string, string> = new Map([
+            ["c", "cache"],
+            ["g", "graph"],
+          ]);
+
+        const fluidKeyValuesToLog: LoggingHeader[] = [
+            { headerName: "Origin", logName: "responseOrigin", mapping: mapResponseOrigin },
+        ];
+
+        const keyValueMap = fluidTelemetry.split(",").map((keyValuePair) => keyValuePair.split("="));
+        for (const [key, value] of keyValueMap) {
+            const fluidKV = fluidKeyValuesToLog.find((t) => t.headerName === key.trim());
+            if (fluidKV !== undefined) {
+                const fieldValue = value?.trim() ?? true; // assume true for a key without value.
+                additionalProps[fluidKV.logName] = fluidKV.mapping?.get(fieldValue) ?? fieldValue;
```

Comment:
I'm lost on this line. :)
dKV.mapping?.get(fieldValue) will give us eiter "cache" or "graph", i.e. pre-computed names.
fieldValue will give as actual values.
I do not get why it's Ok to use them interchangeably here (i.e. occasionally record one or another)

Reply:
Yes, the idea is to use the field value as default unless we have a mapping, like we do for c/cache g/graph. 
Eventually we can ask ODSP folks to add new fields and they do not need to have a mapping. Does this help ?

Reply:
I'd say it is very hard to parse. And I'm not sure it's clear to readers when and why certain values are used. I think this is more readable:
```ts
const keyValueMap = fluidTelemetry.split(",").map((keyValuePair) => keyValuePair.split("="));
for (const [key, value] of keyValueMap) {
	if ("Origin" === key.trim()) {
		let value: string;
		switch (value?.trim()) {
			case "c": 
				value = "cache";
				break;
			case "g": 
				value = "graph";
				break;
			default:
				value = true;
		}
		additionalProps["responseOrigin"] = value;
	}
}
```
I'd not try to optimize for the next case like that - it's very unlikely to ever exists, and even if it does, I'd still prefer easier to follow code. And it's actually less code, and way easier to follow the intention.

I'm not sure that "true" is the right type here. I'd rather write "undefined" (as a string) or some other string that better represents intention. Maybe not recording it at all (i.e. using actual undefined) is Ok, given that we did not learn anything, so it's the case as Origin was not provided at all.